.github/workflows/doxygen.yml

```diff
@@ -11,36 +11,35 @@ jobs:
     runs-on: ubuntu-latest
     container:
       image: novelrt/novelrt-build:latest
-      env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
       - name: Checkout Code
         uses: actions/checkout@v2
 
       - name: Create documentation branch
         run: git checkout -b documentation/$GITHUB_SHA
+        
+      - name: Prepare Secrets
+        run: echo "${{ secrets.GITHUB_TOKEN }}" >> $GITHUB_WORKSPACE/secret.txt
 
       #Mandatory step it appears otherwise pushing the commit will fail
       #Switched username and e-mail to something identifiable?
       - name: Configure Git
         run: |
-          gh auth login
+          gh auth login --with-token < $GITHUB_WORKSPACE/secret.txt
```

Comment:
So how does this work?

Reply:
We currently write the `GITHUB_TOKEN` secret that the workflow has to a file in the temp workspace created by the run - this way we aren't trying to pipe in a string (as `gh cli` _rees like no tomorrow when I do this_). 

Then, we pipe the token into the command following the CLI's instructions (as documented) - this is because `GITHUB_TOKEN` isn't available to our docker container, so it acts like it's a normal computer/VM instead of a specific GH-hosted runner. Because it's using the token from the Action, though, it's able to get the proper permissions.

Please note, we don't publish the `secret.txt` file, and in the runs it is censored as `***` so it isn't truly exposed afaik.